src/Stan/Pattern/Ast.hs

```diff
@@ -79,6 +81,8 @@ data PatternAst
     | PatternAstAnd !PatternAst !PatternAst
     -- | Negation of pattern. Should match everything except this pattern.
     | PatternAstNeg !PatternAst
+    -- | AST node with the specified Identifier details (only 'DeclType')
+    | PatternAstIdentifierDetailsDecl !DeclType
```

Comment:
This is very useful for now :+1: If in the future we will need something more sophisticated or flexible, can always enhance our data type.